backend/src/services/release.ts

```diff
@@ -295,6 +326,168 @@ export async function getReleaseBySemver(user: UserInterface, modelId: string, s
   return release
 }
 
+function getSemverQueryBounds(querySemver: string) {
+  const semverRangeStandardised = semver.validRange(querySemver, { includePrerelease: false })
+
+  if (!semverRangeStandardised) {
+    throw BadReq(`Semver range, '${querySemver}' is invalid `)
+  }
+
+  const [expressionA, expressionB] = semverRangeStandardised.split(' ')
+
+  let lowerSemver: string = '',
+    upperSemver: string = '',
+    lowerInclusivity = false,
+    upperInclusivity = false
+  if (expressionA.includes('>')) {
+    if (expressionA.includes('=')) {
+      lowerSemver = expressionA.replace('>=', '')
+      lowerInclusivity = true
+    } else {
+      lowerSemver = expressionA.replace('>', '')
+    }
+    //do check for expressionB
+    if (expressionB) {
+      if (expressionB.includes('=')) {
+        upperSemver = expressionB.replace('<=', '')
+        upperInclusivity = true
+      } else {
+        upperSemver = expressionB.replace('<', '')
+      }
+    }
+  } else {
+    if (expressionA.includes('=')) {
+      upperSemver = expressionA.replace('<=', '')
+      upperInclusivity = true
+    } else {
+      upperSemver = expressionA.replace('<', '')
+    }
+  }
+
+  let lowerSemverObj: { metadata?: string | undefined; major: number; minor: number; patch: number } = {
+      major: NaN,
+      minor: NaN,
+      patch: NaN,
+    },
+    upperSemverObj: { metadata?: string | undefined; major: number; minor: number; patch: number } = {
```

Comment:
Although JS allows us to define multiple variables on one line like you're doing here, I would strongly advise against it as it's simply less readable than defining multiple variables separately.

```
let lowerSemverObj: SemverObject = { ... }
let upperSemverObj: SemverObject = { ... }
```

Reply:
Fixed above